plugins/hadoop-authorization-sentry-default/sentry-client/src/main/java/com/thinkbiganalytics/datalake/authorization/groups/unix/UnixGroupList.java

```diff
@@ -0,0 +1,116 @@
+package com.thinkbiganalytics.datalake.authorization.groups.unix;
+
+/*-
+ * #%L
+ * thinkbig-sentry-client
+ * %%
+ * Copyright (C) 2017 ThinkBig Analytics
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import com.thinkbiganalytics.datalake.authorization.client.SentryClientConfig;
+import com.thinkbiganalytics.datalake.authorization.model.HadoopAuthorizationGroup;
+import com.thinkbiganalytics.datalake.authorization.model.SentryGroup;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UnixGroupList {
+
+    private static final Logger log = LoggerFactory.getLogger(UnixGroupList.class);
+    private static final String FILENAME = "/etc/group";
+    private String OWNER = "kylo";
+    private String DESCRIPTION = "Kylo Authorization Group";
+    private List<String> groupList ;
+    private List<String> groupId ;
+    private HashMap<String , String> groupInfo ;
+    SentryGroup sentryGroup ;
+    public void  generateGroupList(String groupFilePath)
+    {
+        String[] splitedUnixGroup ;
+        BufferedReader bufferedReader = null;
+        groupList = new ArrayList<>();
+        groupId = new ArrayList<>();
+        groupInfo = new HashMap<>();
+
+        try 
+        {
+            bufferedReader = new BufferedReader(new FileReader(groupFilePath));
+            StringBuilder stringBuilder = new StringBuilder();
+            String line = bufferedReader.readLine();
+
+            if (line == null) {
+                throw new RuntimeException("Unable to generate groups - Linux group file is empty." );
+            }
+
+            while ((line = bufferedReader.readLine()) != null) {
+
+                stringBuilder.append(line);
+
+                if(line.startsWith("#") || line.isEmpty())
+                {
+                    continue;
+                }
+
+                splitedUnixGroup = line.split(":");
+
+                if(splitedUnixGroup.length < 3)
+                {
+                    throw new RuntimeException("Invalid Unix Group Format. Unable to parse " +groupFilePath);

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)